src/ServiceControl.Audit/Infrastructure/RavenDB/Expiration/ExpiredDocumentsCleaner.cs

```diff
@@ -16,6 +16,8 @@ public static Task<TimerJobExecutionResult> RunCleanup(int deletionBatchSize, Do
 
             logger.Debug("Trying to find expired ProcessedMessage and SagaHistory documents to delete (with threshold {0})", threshold.ToString(Default.DateTimeFormatsToWrite, CultureInfo.InvariantCulture));
             AuditMessageCleaner.Clean(deletionBatchSize, database, threshold, token);
+            logger.Debug("Trying to find expired KnownEndpoints to delete (with threshold {0})", threshold.ToString(Default.DateTimeFormatsToWrite, CultureInfo.InvariantCulture));
+            KnownEndpointsCleaner.Clean(deletionBatchSize, database, threshold, token);
```

Comment:
I wonder if it's wise to execute this _every_ time, as in most systems this data (while updating the LastSeen parameter) should remain relatively static. Maybe it should have a separate schedule, or run every 10th time?

Reply:
I saw no reason to diverge from the cleaner pattern we also use in the primary instance https://github.com/Particular/ServiceControl/blob/5163de59031cfea1fbd31e4352b8af0375c9ef53/src/ServiceControl/Infrastructure/RavenDB/Expiration/ExpiredDocumentsCleaner.cs#L13
Right now it is assumed that the expiry interval and settings apply to all cleanups. Given that it is an async process that runs on a customer defined schedule I saw no problem and it keeps the data consistent with the expiry of processed messages so essentially mimicing more or less the current behaviour.

Reply:
The collection is also likely to be relatively small so the overhead of checking it will be negligible compared to the audit cleanup